### PR TITLE
Pipeline tests for Json and Search Modules 

### PR DESCRIPTION
**Pull request by AvitalFineRedis**

### CODE REVIEW

**Review by sazzad16 — CHANGES_REQUESTED** [review]
Create only one test class in src/test/java/redis/clients/jedis/modules/

File: src/test/java/redis/clients/jedis/modules/RedisModulesPipelineTest.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    @BeforeClass
+    public static void prepare() {
+        RedisModuleCommandsTestBase.prepare();
+    }
+
+    @Test
+    public void search() {
+        Jedis jedis = new Jedis(hnp, DefaultJedisClientConfig.builder().timeoutMillis(500).build());
+        jedis.flushAll();
+
```
Comment (sazzad16) [inline src/test/java/redis/clients/jedis/modules/RedisModulesPipelineTest.java:30]:
```suggestion
```

https://github.com/redis/jedis/blob/d45e0cf318bc1f95703636ed3d32b2245d298642/src/test/java/redis/clients/jedis/modules/RedisModuleCommandsTestBase.java#L36-L42

should cover it.

File: src/test/java/redis/clients/jedis/modules/RedisModulesPipelineTest.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        assertEquals(Long.valueOf(1), delResponse.get());
+        assertEquals(99, searchResultResponse2.get().getTotalResults());
+    //    assertEquals("OK", dropIndexResponse.get());
+    }
+
+    @Test
+    public void json() {
+        Jedis jedis = new Jedis(hnp, DefaultJedisClientConfig.builder().timeoutMillis(500).build());
+        jedis.flushAll();
+
```
Comment (sazzad16) [inline src/test/java/redis/clients/jedis/modules/RedisModulesPipelineTest.java:65]:
```suggestion
```

https://github.com/redis/jedis/blob/d45e0cf318bc1f95703636ed3d32b2245d298642/src/test/java/redis/clients/jedis/modules/RedisModuleCommandsTestBase.java#L36-L42

should cover it.

**Review by sazzad16 — CHANGES_REQUESTED** [review]
We need more methods to be covered.

File: src/test/java/redis/clients/jedis/modules/RedisModulesPipelineTest.java (hunk 1 above)
Comment (AvitalFineRedis) [reply src/test/java/redis/clients/jedis/modules/RedisModulesPipelineTest.java:30]: It's not :sweat_smile: , client (UnifiedJedis) does not have Pipeline

File: src/main/java/redis/clients/jedis/Pipeline.java [hunk 3]
```diff
@@ -2800,7 +2800,7 @@ public final boolean hasPipelinedResponse() {
   }
 
   @Override
-  public Response<Object> jsonGet(String key) {
+  public Response<Object> jsonGet(String key) {
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/Pipeline.java:2803]: undo

File: src/main/java/redis/clients/jedis/Pipeline.java [hunk 4]
```diff
@@ -2855,7 +2855,7 @@ public final boolean hasPipelinedResponse() {
   }
 
   @Override
-  public Response<Long> jsonClear(String key, Path2 path) {
+  public Response<Long> jsonClear(String key, Path2 path) {
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/Pipeline.java:2858]: undo

File: src/main/java/redis/clients/jedis/Pipeline.java [hunk 5]
```diff
@@ -2925,7 +2925,7 @@ public final boolean hasPipelinedResponse() {
   }
 
   @Override
-  public Response<List<Long>> jsonArrAppendWithEscape(String key, Path2 path, Object... objects) {
+  public Response<List<Long>> jsonArrAppendWithEscape(String key, Path2 path, Object... objects) {
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/Pipeline.java:2928]: undo

**Review by sazzad16 — CHANGES_REQUESTED** [review]

**Review by sazzad16 — APPROVED** [review]

**Pull request merged**